crates/uv/tests/it/publish.rs

```diff
@@ -197,3 +201,175 @@ fn dubious_filenames() {
     "###
     );
 }
+
+/// Check that we (don't) use the keyring and warn for missing keyring behaviors correctly.
+#[test]
+fn check_keyring_behaviours() {
+    let context = TestContext::new("3.12");
+
+    // No upper bound, since we are recommending `uv tool install keyring` without upper bound to
+    // users. The keyring package had a bad release in the past, but so it hits us before it hits
+    // users.
```

Comment:
I want to avoid unbounded dependencies in the test suite for security. Can we continuously test against the latest keyring in a dedicated CI job instead so it doesn't run on developer machines?